### ISSUE TITLE
NVDA reads tooltips although this is disabled

### ISSUE THREAD

**Issue by spanky-acapulco** [nvaccess/nvda#9480]
### Steps to reproduce:
Make sure, *Settings > Object Presentation > Report Tool Tips* is **disabled** in your NVDA.
Open this in Firefox with NVDA, **focus one of the links**. 
https://codepen.io/martin-mengele/pen/qwrpKK

### Actual behavior:
NVDA reads the link's name and it's description.
Nothing too conspicuous in the Firefox A11y tree, see image.

![image](https://user-images.githubusercontent.com/3273686/55891731-5d0f4200-5bb5-11e9-8ca3-82804a41cedd.png)
![image](https://user-images.githubusercontent.com/3273686/55891783-76b08980-5bb5-11e9-9c29-0cebbdc6912d.png)

### Expected behavior:
It should only read the link's name.

#### NVDA installed/portable/running from source:
NVDA installed

#### NVDA version:
2019.1
#### Windows version:
Windows 10
#### Name and version of other software in use when reproducing the issue:
Firefox 66.0.2 (64-Bit)
#### Does the issue still occur after restarting your PC?
Yes
#### Have you tried any other versions of NVDA?
Yes, 2018.4, same browser. Same result.

Hello NVDA Team,
Thank you for your incredible work! I hope I can help you making NVDA even better!
Greets from Germany,
Martin

**Comment by feerrenrut** [comment]
Thank you for the detailed bug report, however please be aware that many of the people who maintain NVDA are blind or vision impaired. Some of us are helped by images, though it's best if accurate text descriptions are provided first.

The title attribute is being mapped to the accessible description, [this is expected](https://www.w3.org/TR/WCAG20-TECHS/H33.html) to be announced.

Using Chrome, with your code pen. You can see that the title is exposed as the accessible description of the element. In this case the tool tip shown when hovering the mouse was not being shown.
```
Developer info for navigator object:
name: u'This is my link name 2'
role: ROLE_LINK
states: STATE_FOCUSABLE, STATE_LINKED, STATE_FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.IAccessible.ia2Web.Ia2Web object at 0x055B0F90>
Python class mro: (<class 'NVDAObjects.IAccessible.ia2Web.Ia2Web'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: u'This is my title 2'
location: RectLTWH(left=315, top=465, width=288, height=36)
value: u'https://s.codepen.io/martin-mengele/fullpage/qwrpKK#'
appModule: <'appModuleHandler' (appName u'chrome', process ID 2672) at address 5469f30>
appModule.productName: u'Google Chrome'
appModule.productVersion: u'73.0.3683.86'
TextInfo: <class 'NVDAObjects.IAccessible.IA2TextTextInfo'>
windowHandle: 1968304
windowClassName: u'Chrome_RenderWidgetHostHWND'
windowControlID: 209374480
windowStyle: 1445986304
windowThreadID: 12480
windowText: u'Chrome Legacy Window'
displayText: u''
IAccessibleObject: <POINTER(IAccessible2) ptr=0x521c674 at 591f5d0>
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=1968304, objectID=-4, childID=-107485
IAccessible accName: u'This is my link name 2'
IAccessible accRole: ROLE_SYSTEM_LINK
IAccessible accState: STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_LINKED, STATE_SYSTEM_VALID (5242884)
IAccessible accDescription: u'This is my title 2'
IAccessible accValue: u'https://s.codepen.io/martin-mengele/fullpage/qwrpKK#'
IAccessible2 windowHandle: 1968304
IAccessible2 uniqueID: -107485
IAccessible2 role: ROLE_SYSTEM_LINK
IAccessible2 states: IA2_STATE_OPAQUE (1024)
IAccessible2 attributes: u'display:inline;tag:a;'
```

Testing tool tip reporting myself (in chrome):
1. With "report tool tips" is turned on. (Mouse tracking off, for simplicity)
1. Move the mouse over the link, wait for the tooltip to be displayed
1. The following is announced: "This is my title 2  tool tip"
1. Turn off "report tool tips"
1. Move the mouse over the link, wait for the tooltip to be displayed.
1. Nothing is announced.

In Firefox I could not get the tooltip to be reported.

I'm going to close this issue, since I can not reproduce it. However reading between the lines here:
- I'm guessing there is a desire to be able to present a tooltip (using the title attribute) which does not add useful information?
- If it is useful to people without a screenreader, why is it not useful to those with a screenreader?

**Comment by spanky-acapulco** [comment]
Sorry for leaving out the image alt texts, but I could not find a way to add them (my first issue so far). And thanx for immediate reply!

To your test: I disabled the report tooltips option, but it reads the tooltip on focussing the link, although it's not displayed or anything. This is not expected behaviour I think, if I disable this in NVDA. No?

**Comment by DrSooom** [comment]
@spanky-acapulco: Are you moving the cursor to a link in browse or in focus mode? You can toggle between them by pressing NVDA+Space. Use Tab in focus mode and K/Shift+K in browse mode.

**Comment by spanky-acapulco** [comment]
Although I didn't switch the modes I just tried this in different modes:

### browse mode
- moving to the link via tab/shift+tab (focusing the link): it reads tool tips (title attribute)
- moving the cursor via ↑,↓ keys: it doesn't read tool tips (title attribute)

### focus mode: 
- moving to the link via tab/shift+tab (focusing the link): it reads tool tips (title attribute)

![Screenshot: settings menu in NVDA (German Version!): Object Presentation > Report Tool Tips is disabled](https://user-images.githubusercontent.com/3273686/55936282-1a417e80-5c36-11e9-97ea-ec2b1e0a28f6.png)

Maybe I'm getting this setting wrong: I assumed that NVDA is supposed to NOT read tool tips at all. But I think this setting is only valid, if you use a mouse as well, then it doesn't read the tool tip on mouse over.

Sorry for the inconvenience and greets from Germany,
Martin

**Comment by feerrenrut** [comment]
> Sorry for leaving out the image alt texts, but I could not find a way to add them 

No alt text necessary, just write text before the image, eg "the following image shows: blah"

I think the confusion comes from a misunderstanding: The title attribute is not equal to a tool tip.

I believe this is the expected behaviour:
- if an application shows a tooltip (a little box with text in it that appears over the top of the application), then it is read with "tool tip" appended.
- This generally happens when you put the mouse over the element

An image with the code pen example, a tooltip showing, and speechviewer open reporting the tooltip
![image](https://user-images.githubusercontent.com/3491346/55956972-7cfe3e80-5c65-11e9-9287-b4d5faa6eaaa.png)

And after turning off report tool tips.
An image with the code pen example, a tooltip showing, and speechviewer open NOT reporting the tooltip 
![image](https://user-images.githubusercontent.com/3491346/55957348-686e7600-5c66-11e9-9aa9-9360eca6d50b.png)

In your examples:
- "tool tip" is not appended
- I assume the tool tip is also not visible
- I assume that you are moving focus to the element with the tab key or down arrow.

If I do this:
An image with the code pen example, focus on the second link, and speechviewer open NOT reporting the tooltip, but showing the link text, and the title text. (Accessible name and accessible description)
![image](https://user-images.githubusercontent.com/3491346/55957514-d5820b80-5c66-11e9-929f-1b1c7a6b3e07.png)

The `title` attribute is being announced.
 - It is exposed by the browser as the accessible description of the element.
- This is expected behaviour

**Comment by spanky-acapulco** [comment]
Yep, I got it right now. Thanx for the clarification!

**Comment by ppathania123** [comment]
NVDA is not reading tooltips in Dynamics CRM application.
Any help on this?

**Comment by DrSooom** [comment]
@ppathania123: Please open a new issue for this. It seems that your question has nothing to do with the issue here. Thanks.
